projects/plugins/crm/src/js/components/automations-admin/components/attribute-config/index.tsx

```diff
@@ -55,13 +65,30 @@ const getEditValue = (
 				/>
 			);
 		case 'text':
-			return <TextControl value={ value.toString() } onChange={ onChange } />;
+			return <TextControl value={ value } onChange={ onChange } />;
 		case 'checkbox':
+			return (
+				<CheckboxControl
+					label={ definition.title }
+					checked={ value as boolean }
+					onChange={ onChange }
+				/>
+			);
 		case 'textarea':
+			return <TextareaControl value={ value as string } onChange={ onChange } />;
 		case 'date':
-		case 'datetime':
+			return (
+				<div className={ styles.datepicker }>
+					<DatePicker
+						currentDate={ value ? new Date( value as string | number ) : new Date() }
+						onChange={ ( selectedDate: Date ) => {
+							onChange( selectedDate.getTime() );
+						} }
```

Comment:
This isn't updating Redux for me because `selectedDate` is a string.

```suggestion
						onChange={ ( selectedDate: string ) => {
							onChange( selectedDate );
						} }
```

Reply:
Ah! Great catch! It should be fixed now by passing the string to a new `Date` and then converting it to a timestamp 🙈